components/Form.js

```diff
@@ -2,17 +2,24 @@ import styled from 'styled-components';
 import Image from 'next/image';
 import Logo from '../public/Logo.png';
 import { useState } from 'react';
+import calculation from './utils/calculation';
 
-export default function Form({ onCalculatePrice, value }) {
+export default function Form({ coins }) {
+  const [value, setValue] = useState();
+  const [selectedCoinName, setSelectedCoinName] = useState('');
+  const [disableButton, setdisableButton] = useState(true);
   function handleSubmit(event) {
     event.preventDefault();
     const form = event.target;
-    const expenses = Number(form.expenses.value);
+    const expense = Number(form.expenses.value);
     const cryptovalue = Number(form.cryptovalue.value);
+    const selectedCoin = coins.filter((coin) => coin.name == selectedCoinName);
 
-    onCalculatePrice(cryptovalue, expenses);
+    const calculatedMonths = calculation(cryptovalue, expense, selectedCoin[0]);
```

Comment:
```suggestion
    const calculatedMonths = calculation(cryptovalue, expense, selectedCoin);
```
No array index needed if you use `find()` before instead of `filter()`.

Reply:
Thank you for the clarification. Have changed it